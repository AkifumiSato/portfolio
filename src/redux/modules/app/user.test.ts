import * as snapshotDiff from 'snapshot-diff'
import reducer, {
  UserModel,
  updateNameAction,
  updateEmailAction,
  updateCommentAction,
  validateAction
} from './user'

const initialState = new UserModel()

test('[app user state]: init', () => {
  expect(
    snapshotDiff(undefined, reducer(undefined, { type: '@@INIT' }))
  ).toMatchSnapshot()
})

// name
test('[app user state]: update name', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateNameAction('test')))
  ).toMatchSnapshot()
})

test('[app user state]: empty name', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateNameAction('')))
  ).toMatchSnapshot()
})

test('[app user state]: too long name', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateNameAction('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')))
  ).toMatchSnapshot()
})

// email
test('[app user state]: update email', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateEmailAction('test@test.com')))
  ).toMatchSnapshot()
})

test('[app user state]: empty email', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateEmailAction('')))
  ).toMatchSnapshot()
})

test('[app user state]: error email', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateEmailAction('test@test')))
  ).toMatchSnapshot()
})

test('[app user state]: too long email', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateNameAction('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa@test.com')))
  ).toMatchSnapshot()
})

// comment
test('[app user state]: update comment', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateCommentAction('test')))
  ).toMatchSnapshot()
})

test('[app user state]: empty comment', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateCommentAction('')))
  ).toMatchSnapshot()
})

test('[app user state]: too long comment', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, updateCommentAction('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')))
  ).toMatchSnapshot()
})

// validation
test('[app user state]: validation', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, validateAction()))
  ).toMatchSnapshot()
})
