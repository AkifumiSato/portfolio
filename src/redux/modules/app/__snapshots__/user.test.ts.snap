// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[app user state]: empty comment 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,9 +6,9 @@
    \\"email\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },
    \\"comment\\": Object {
-     \\"error\\": \\"\\",
+     \\"error\\": \\"コメントは必須です。\\",
      \\"value\\": \\"\\",
    },
  }"
`;

exports[`[app user state]: empty email 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,11 +2,11 @@
    \\"name\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },
    \\"email\\": Object {
-     \\"error\\": \\"\\",
+     \\"error\\": \\"メールアドレスは必須です。\\",
      \\"value\\": \\"\\",
    },
    \\"comment\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\","
`;

exports[`[app user state]: empty name 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  Immutable.Record {
    \\"name\\": Object {
-     \\"error\\": \\"\\",
+     \\"error\\": \\"名前は必須です。\\",
      \\"value\\": \\"\\",
    },
    \\"email\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\","
`;

exports[`[app user state]: error email 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,12 +2,12 @@
    \\"name\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },
    \\"email\\": Object {
-     \\"error\\": \\"\\",
-     \\"value\\": \\"\\",
+     \\"error\\": \\"メールアドレスが不正です。\\",
+     \\"value\\": \\"test@test\\",
    },
    \\"comment\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },"
`;

exports[`[app user state]: init 1`] = `
"Snapshot Diff:
  Comparing two different types of values. Expected undefined but received object."
`;

exports[`[app user state]: too long comment 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,9 +6,9 @@
    \\"email\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },
    \\"comment\\": Object {
-     \\"error\\": \\"\\",
-     \\"value\\": \\"\\",
+     \\"error\\": \\"コメントは1000文字までです。\\",
+     \\"value\\": \\"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\\",
    },
  }"
`;

exports[`[app user state]: too long email 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,9 +1,9 @@
  Immutable.Record {
    \\"name\\": Object {
-     \\"error\\": \\"\\",
-     \\"value\\": \\"\\",
+     \\"error\\": \\"名前は100文字までです。\\",
+     \\"value\\": \\"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa@test.com\\",
    },
    \\"email\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },"
`;

exports[`[app user state]: too long name 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,9 +1,9 @@
  Immutable.Record {
    \\"name\\": Object {
-     \\"error\\": \\"\\",
-     \\"value\\": \\"\\",
+     \\"error\\": \\"名前は100文字までです。\\",
+     \\"value\\": \\"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\\",
    },
    \\"email\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },"
`;

exports[`[app user state]: update comment 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,8 +7,8 @@
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },
    \\"comment\\": Object {
      \\"error\\": \\"\\",
-     \\"value\\": \\"\\",
+     \\"value\\": \\"test\\",
    },
  }"
`;

exports[`[app user state]: update email 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -3,11 +3,11 @@
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },
    \\"email\\": Object {
      \\"error\\": \\"\\",
-     \\"value\\": \\"\\",
+     \\"value\\": \\"test@test.com\\",
    },
    \\"comment\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },"
`;

exports[`[app user state]: update name 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,9 +1,9 @@
  Immutable.Record {
    \\"name\\": Object {
      \\"error\\": \\"\\",
-     \\"value\\": \\"\\",
+     \\"value\\": \\"test\\",
    },
    \\"email\\": Object {
      \\"error\\": \\"\\",
      \\"value\\": \\"\\",
    },"
`;

exports[`[app user state]: validation 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Immutable.Record {
    \\"name\\": Object {
-     \\"error\\": \\"\\",
+     \\"error\\": \\"名前は必須です。\\",
      \\"value\\": \\"\\",
    },
    \\"email\\": Object {
-     \\"error\\": \\"\\",
+     \\"error\\": \\"メールアドレスは必須です。\\",
      \\"value\\": \\"\\",
    },
    \\"comment\\": Object {
-     \\"error\\": \\"\\",
+     \\"error\\": \\"コメントは必須です。\\",
      \\"value\\": \\"\\",
    },
  }"
`;
